clippy_lints/src/let_underscore.rs

```diff
@@ -43,8 +76,17 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for LetUnderscore {
             if let PatKind::Wild = local.pat.kind;
             if let Some(ref init) = local.init;
             then {
-                if is_must_use_ty(cx, cx.tables.expr_ty(init)) {
-                   span_lint_and_help(
+                let check_ty = |ty| SYNC_GUARD_PATHS.iter().any(|path| match_type(cx, ty, path));
+                if cx.tables.expr_ty(init).walk().any(check_ty) {
+                        span_lint_and_help(
+                            cx,
+                            LET_UNDERSCORE_LOCK,
+                            stmt.span,
+                            "non-binding let on a synchronization lock",
+                            "consider using an underscore-prefixed named binding or dropping explicitly with `std::mem::drop`"
+                        )
```

Comment:
double indentation

Reply:
Why doesn't fmt do that?

Reply:
rustfmt currently only formats in some macros. The `if_chain` macro is too complex, for rustfmt to format the code in it reliably.